### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by scottaltair**
Added Requirements
Clarified Installation Steps for sparks.pbs.executor.home

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,18 +30,26 @@ bin/spark-class org.apache.spark.deploy.pbs.ui.PbsClusterUI
 ```
 
 
-
 ## Installation
 
-This expects PBSPro to be installed at `/opt/pbs`.
+### Requirements
+1. PBS Professional must be installed in default locaiton, at `/opt/pbs`.
+2. Confirm host resources (CPU, Memory, and Disk) are sufficient for Spark project and build
```
Comment (saimanikant) [inline README.md:37]: Adding the following link here would be really helpful : https://spark.apache.org/docs/latest/hardware-provisioning.html

Comment (scottaltair) [reply README.md:37]: I agree. The original README was very light on information and knowledge.

File: README.md [hunk 2]
```diff
@@ -31,18 +30,26 @@ bin/spark-class org.apache.spark.deploy.pbs.ui.PbsClusterUI
...
 
 
-
 ## Installation
 
-This expects PBSPro to be installed at `/opt/pbs`.
+### Requirements
+1. PBS Professional must be installed in default locaiton, at `/opt/pbs`.
+2. Confirm host resources (CPU, Memory, and Disk) are sufficient for Spark project and build. More information about these resource requirements can be found at https://spark.apache.org/docs/latest/hardware-provisioning.html
+3. Spark project must be accessible by the submission host (PBS Client) and execution host(s) (PBS MoM), which is consistent on all hosts.
```
Comment (sujatapatnaik52) [inline README.md:38]: I believe in HPC clusters, users should be able to do spark-submit from PBS Client machine only. If this is right then, we should ask users to use spark-submit in cluster mode only. With spark-submit in client mode, the spark driver will accumulate resources from the PBS Client machine which I am not sure is correct. Let me know your thoughts if that needs to be updated in README.

File: README.md [hunk 3]
```diff
@@ -21,8 +21,7 @@ You can run Spark on the PBS cluster just by adding "--master pbs" while submitt
 ./bin/spark-submit --master pbs --deploy-mode cluster --class org.apache.spark.examples.SparkPi $SPARK_HOME/examples/target/scala-2.12/jars/spark-examples_2.12-3.1.0-SNAPSHOT.jar 100
```
Comment (sujatapatnaik52) [inline README.md:21]: I know this is not related to your change. Can we update the above `./bin/spark-submit` to `$SPARK_HOME/bin/spark-submit` for both the commands? Thanks.

File: README.md [hunk 4]
```diff
@@ -54,8 +61,15 @@ git am resource-managers/pbs/*.patch
 build/mvn -DskipTests -Ppbs package
 ```
 
-Add executor home to your configuration:
+Update `conf/sparks-default.conf` by adding executor home to your configuration:
 ```bash
 # in file conf/spark-defaults.conf add line:
-spark.pbs.executor.home "SPARK INSTALLATION DIRECTORY PATH IN PBS MOMS"
+spark.pbs.executor.home /common/path/to/spark/folder/on/hosts
```
Comment (sujatapatnaik52) [inline README.md:67]: Looking at the code, I think this is a mandatory config which needs to be set for cluster mode. can we update the same here?

File: README.md (hunk 2 above)
Comment (scottaltair) [reply README.md:38]: The requirement is to have the Spark project. I will update to clarify $SPARK_HOME is accessible by the submission host and execution host(s).

File: README.md (hunk 3 above)
Comment (scottaltair) [reply README.md:21]: Yep. Will update. That makes more sense than what the original author had. I also found using $SPARK_HOME to be more accurate.

File: README.md (hunk 4 above)
Comment (scottaltair) [reply README.md:67]: I don't understand your suggestion. Line 64-67 of README is required. I do NOT suggest this to be optional. 

However, line 70-74 if README is _optional_.

Comment (sujatapatnaik52) [reply README.md:68]: I see now that this config comes under `Requirements:` section. Thank you for clarifying it.

File: README.md (hunk 2 above)
Comment (sujatapatnaik52) [reply README.md]: yeah this requirement in absolutely correct. I was actually going in a different direction where the mode of deployment and spark-submit comes into picture. This is not really a requirement but kind of letting users know that when doing spark-submit from PBS Client machine, please use the deployment mode as cluster i.e. `spark-submit --deploy cluster`. I am not sure if we should add that in README or not.

**Pull request merged**